src/coreclr/jit/simdashwintrinsic.cpp

```diff
@@ -918,16 +925,24 @@ GenTree* Compiler::impSimdAsHWIntrinsicSpecial(NamedIntrinsic       intrinsic,
             if (SimdAsHWIntrinsicInfo::SpillSideEffectsOp1(intrinsic))
             {
                 impSpillSideEffect(true, verCurrentState.esStackDepth -
-                                             2 DEBUGARG("Spilling op1 side effects for SimdAsHWIntrinsic"));
+                                             (newobjThis == nullptr ? 2 : 1)DEBUGARG(
```

Comment:
Why does this need `2` vs `1`? Whether `newobjThis` is there or not, we should still have exactly two operands on the stack that need to be popped and consumed here.

Reply:
`newobjThis` is used when the `NEWOBJ` instruction is used. We don't have the address in the stack here.